### PR TITLE
Support for IME

### PR DESCRIPTION
**Pull request by budnix**
### Context

This PR fixes several issues related to editing cell values for users who use IME as an input method.

### How has this been tested?

This was tested on several OS (macOS, Win 10, Win 7, Ubuntu 17/18) using the most popular browsers (Chrome, FF, Safari, Edge, and IE).

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #4662

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/mixins/localHooks.js [hunk 1]
```diff
@@ -19,12 +19,15 @@ const localHooks = {
    *
    * @param {String} key Hook name.
    * @param {Function} callback Hook callback
+   * @return {Object}
```
Comment (swistach) [inline src/mixins/localHooks.js]: Could you add `s` to `@return`?

File: src/mixins/localHooks.js [hunk 2]
```diff
@@ -41,9 +44,13 @@ const localHooks = {
 
   /**
    * Clear all added hooks.
+   *
+   * @return {Object}
```
Comment (swistach) [inline src/mixins/localHooks.js]: Could you add `s` to `@return`?

File: src/plugins/copyPaste/test/copyPaste.e2e.js [hunk 3]
```diff
@@ -57,6 +59,25 @@ describe('CopyPaste', () => {
...
+    handsontable();
+    selectCell(0, 0);
+
+    await sleep(10);
+
+    expect(document.activeElement).toBe(getActiveEditor().TEXTAREA);
+    expect($('#HandsontableCopyPaste').length).toBe(0);
+  });
+
+  it('should create focusable element when cell editor doesn\'t exist', async () => {
```
Comment (swistach) [inline src/plugins/copyPaste/test/copyPaste.e2e.js]: Is `async` necessary?

File: test/e2e/Core_validate.spec.js [hunk 4]
```diff
@@ -1380,7 +1380,7 @@ describe('Core_validate', () => {
     }, 200);
   });
 
-  it('should close the editor and save the new value after double clicking on a cell, if the previously edited cell validated correctly', (done) => {
+  it('should close the editor and save the new value after double clicking on a cell, if the previously edited cell validated correctly', async () => {
```
Comment (swistach) [inline test/e2e/Core_validate.spec.js]: Are `validated` and `clicks` still necessary?

File: src/plugins/copyPaste/test/copyPaste.e2e.js (hunk 3 above)
Comment (budnix) [reply src/plugins/copyPaste/test/copyPaste.e2e.js]: Fixed

**Comment by dcalde** [comment]
Has this PR been tested with custom editors that are not based on textarea fields? I am using https://github.com/mydea/handsontable-chosen-editor which is throwing an error since upgrading to 4.0.0.

```
handsontable.full.js:76587 Uncaught TypeError: this.mainElement.select is not a function
    at FocusableWrapper.focus (handsontable.full.js:76587)
    at CopyPaste.onAfterSelectionEnd (handsontable.full.js:76312)
```

mainElement == <select>

**Comment by AMBudnik** [comment]
Hi @dcalde 

can you share the code of your editor or maybe it is [this one](https://github.com/dcalde/handsontable-chosen-editor/blob/master/handsontable-chosen-editor.js)? It will help us to improve the IME fix.

**Pull request merged**